kii_thing_if.c

```diff
@@ -684,6 +686,30 @@ static kii_bool_t prv_onboard_with_vendor_thing_id(
             return KII_FALSE;
         }
     }
+    if (firmware_version != NULL) {
+        if (kii_api_call_append_body(kii, ",\"firmwareVersion\":",
+                        CONST_STRLEN(",\"firmwareVersion\":")) != 0) {
+            M_KII_LOG(kii->kii_core.logger_cb("request size overflowed.\n"));
+            return KII_FALSE;
+        }
+        if (kii_api_call_append_body(kii, firmware_version,
```

Comment:
I think your opinion is following code is better than current code. Right?

```c
    if (layout_position != NULL) {
        if (kii_api_call_append_body(kii, ",\"layoutPosition\":",
                        CONST_STRLEN(",\"layoutPosition\":")) != 0 ||
                kii_api_call_append_body(kii, layout_position,
                        strlen(layout_position)) != 0) {
            M_KII_LOG(kii->kii_core.logger_cb("request size overflowed.\n"));
            return KII_FALSE;
        }
    }
```

I think the condition of `if` statement is complex to read.
What do you think about following code?

```c
    if (layout_position != NULL) {
        if (append_key_value(
                kii,
                "layoutPosition", CONST_STRLEN("layoutPosition"),
                layout_position, strlen(layout_position)) != 0) {
            M_KII_LOG(kii->kii_core.logger_cb("request size overflowed.\n"));
            return KII_FALSE;
        }
    }
```

I think this is more easy to read.
What do you think?